### ISSUE TITLE
Next.js 15 + Expo 52 + React 19 (✅)

### ISSUE THREAD

**Issue by edisonhuff** [nandorojo/solito#498]
Heyo, wondering if there's any breaking changes I need to worry about when using solito with expo 52, and if there will be an explicit update for supporting expo 52.

**Comment by nandorojo** [comment]
There shouldn’t be any issues since this library is JS only. The bigger thing to watch for would be upgrading to React Navigation v7, but I think that should work too. Just need to confirm it.

SDK 52 should have no affect on Solito

**Comment by nandorojo** [comment]
That looks like an expo router error in particular. There might be clashing versions of react navigation sub packages, be sure it’s not in package json

**Comment by nandorojo** [comment]
Maybe duplicate react versions, a yarn resolution could fix this

**Comment by nandorojo** [comment]
Ah, got it, merged

**Comment by nandorojo** [comment]
I actually created the upstream PR from react navigation for LinkingContext too haha

**Comment by megacherry** [comment]
No error, only routing does not work with useRouter() -> ```() => router.push('/settings')``` does nothing. No error message, nothing. But if i replace it with expo-router, then it works. Could it be react-navigation v6?

**Comment by nandorojo** [comment]
Hm it could be some bifurcation between Expo Router/React Navigation.

If I saw a reproduction it might be easier to say. We’re using useLinkTo from react navigation, does that work?

**Comment by megacherry** [comment]
Hm, useLinkTo is working...I'm looking further.

**Comment by nandorojo** [comment]
Can you make sure you don’t have multiple versions of react navigation installed

**Comment by nandorojo** [comment]
Closing this, please open a separate issue if needed. For anyone who finds this issue, please upgrade Solito 4.3.0

**Comment by nandorojo** [comment]
Release here: https://github.com/nandorojo/solito/releases/tag/v4.3.0

**Comment by megacherry** [comment]
Got it. It's an useLayoutEffect with navigation.setOptions.

This is actually not working (why ever):

```
React.useEffect(() => {
    navigation.setOptions({
      title: value === '' ? 'No title' : value,
    });
  }, [navigation, value]);
```

**Comment by nandorojo** [comment]
That looks unrelated to this issue (and Solito), could you open an issue on react navigation?

**Comment by garygcchiu** [comment]
FWIW I'm running into the same issue as @megacherry. Upgraded to Expo52, updated all my dependencies (including solito@4.3.0), and now my navigation does not work. Looking into my yarn.lock I should only have react-navigation ^7.0.0. Navigating via expo-router (i.e. my Tab navigation) works fine, however. Also investigating 🤔

**Comment by garygcchiu** [comment]
Alright I found something interesting, this navigation issue only happens if I have a Stack.Screen defined with my route. For example: https://github.com/mozzius/graysky/blob/9d4825ad66058c2127e049093d3f6accbac1a7d8/apps/expo/src/app/(tabs)/(feeds%2Csearch%2Cnotifications%2Cself)/feeds/manage.tsx#L22-L57. 

I _believe_ the alternative here would be defining this Stack.Screen in an `_layout.tsx` file instead.

**Comment by megacherry** [comment]
Yeah, i had the same result. The routing worked without Stack.Screen. But I gave up and downgraded to expo sdk 51 again.

**Comment by nandorojo** [comment]
Good to know, is the issue due to Expo Router then?

Based on https://github.com/expo/expo/issues/32970, it looks like it's due to React Native Screens! The depth of dependency hell never ends haha

**Comment by notdotscott** [comment]
Are we any closer to getting Expo 52 support? Expo will be releasing SDK 53 in "spring" and Solito risks falling even further behind 😢

**Comment by nandorojo** [comment]
I’ve actually been using it in SDK 52 just fine…the issue is with Expo Router, right? I can look into that today. Could you send me your versions you’re using so I can reproduce please?

Also @garygcchiu the broken code you shared here is simply invalid React Navigation code. You have to pass components inline: https://github.com/nandorojo/solito/issues/498#issuecomment-2477689825

**Comment by nandorojo** [comment]
Either way, I’ll be sure to ensure a working version gets out if it’s not working

**Comment by notdotscott** [comment]
@nandorojo I was actually trying with the Expo Router starter. When I upgraded Expo to SDK 52 I ran into the same issue as @garygcchiu. The starter is using SDK 49 by default and I found that I could update to SDK 51 without issue, but not SDK 52. 

My project is a few months old now and trying to run it with Expo Go no longer works due to what seems like an unrelated issue. Let me put together another example from scratch and I can give you some more detail.

**Comment by nandorojo** [comment]
Great thanks. As much info on dependencies / minimal repo helps 🙏

**Comment by notdotscott** [comment]
@nandorojo minimal repo available here: https://github.com/notdotscott/solito-expo-router-sdk52

All I've done is cloned the Expo Router starter via the CLI instructions here: https://solito.dev/expo-router and updated all the relevant dependencies; main one was Expo SDK to 52 and then all the various other Expo dependencies listed when running `npx expo-doctor@latest`. I also updated `metro` and `metro-resolver` in the root package.json.

The code runs and I can load the app using Expo Go. However, the navigation between pages breaks as per the other reports in this thread:

- Click "Regular Link" to go the the user page
- Click "Go Home" to go back to the home page
- Click "Regular Link" again to go to the user page
- Observe that the top "nav" bar disappears and the "Go Home" link no longer works - user is stuck in the user page. 

There are no errors logged to the terminal or to React Native DevTools. No app related code was changed (please see the only changes that were made via this commit: https://github.com/notdotscott/solito-expo-router-sdk52/commit/422c401e357f6f6f04e79629abe72a92934dc70e)

**Comment by nandorojo** [comment]
Working on Next.js 15 and then I'll look at this next

**Comment by nandorojo** [comment]
New Compatibility and upgrade docs here: 

https://solito.dev/compatibility

**Comment by notdotscott** [comment]
@nandorojo this is great, thank you!

**Comment by hbhanoo** [comment]
@nandorojo I'm super new to this whole stack but I'm getting an error when either trying to create a new blank app from `npx create-solito-app@latest` or from running yarn native from within the `blank` example monorepo:

```
iOS Bundled 564ms apps/expo/index.js (1151 modules)
 (NOBRIDGE) LOG  Bridgeless mode is enabled
 INFO
 � JavaScript logs will be removed from Metro in React Native 0.77! Please use React Native DevTools as your default tool. Tip: Type j in the terminal to open (requires Google Chrome or Microsoft Edge).
 (NOBRIDGE) ERROR  Error: Failed to parse the href to a navigation state.
```

![Image](https://github.com/user-attachments/assets/0c338f74-62e9-4eb8-bbe9-4c5ac8996399)

**Comment by nandorojo** [comment]
Hmm this would happen (I think) if you haven't added a screen to your linking config in the navigation provider

**Comment by nandorojo** [comment]
This happens in a blank Solito starter for you? Like no new code?

**Comment by hbhanoo** [comment]
> This happens in a blank Solito starter for you? Like no new code?

correct. 
```
$ git st
On branch master
Your branch is up to date with 'origin/master'.

nothing to commit, working tree clean
$ yarn native
...
```
then scan QR code and click on one of the links.

**Comment by MattKevan** [comment]
Yeah I'm getting the same problem - brand new Solito install. The web version works fine, but on native I'm getting an error when clicking on any of the links:

```
 (NOBRIDGE) ERROR  Error: Failed to parse the href to a navigation state.
 (NOBRIDGE) ERROR  Error: Failed to parse the href to a navigation state.
```

This is using the Expo Go app on iOS.

**Comment by Tobi696** [comment]
Because there is a typo/mistake in the blank monorepo. There is one route called user and another one user**s**

I will create a PR with working version including nativewind v4 and rnr soon.

**Comment by Tobi696** [comment]
Please check if the with-tailwind template works for you when copied from my PR.
https://github.com/nandorojo/solito/pull/506

**Comment by nandorojo** [comment]
Hey, I think this should fix it! There was a small typo in the linking config:

https://github.com/nandorojo/solito/commit/9d871134d45da5db34e127818847c741817bbc4f

**Comment by nandorojo** [comment]
@hbhanoo can you try that and let me know please?

**Comment by MattKevan** [comment]
Wow that was quick! That worked, thanks for your help.

**Comment by hbhanoo** [comment]
Thanks for the quick turnaround! Wasn't sure if my tooling was wrong since I'm just getting started.
